blog/2015/11/swdcretreat2015-university_course.html

```diff
@@ -0,0 +1,436 @@
+
+<p>
+  <h4>
+    Earlier this year, Anthony Scopatz and Kathryn Huff published <a href='http://physics.codes/'>Effective Computation in Physics</a> and created a course around the book.
+    Tiffany, can you talk about the <a href='https://phy3009.github.io/PHY3009-2015/'>course</a> you've been beta testing?
+  </h4>
+</p>
+
+<p>
+  [T]: Anthony has been teaching the course to Masters students and I've been building off Katy and Anthony's experiences from teaching as well as my own from teaching Software Carpentry.
+  I decided to use this textbook as my course at Quest.
+  The book is based on Python, but many of the principles are language agnostic.
+  The book covers many topics SWC teaches, such as Bash, Python, Make, documentation, licenses, and many open science aspects.
+  Katy and Anthony also created a <a href='https://github.com/physics-codes/examples'>Github repo</a> of annotated iPython (Jupyter) notebooks of the code in the book, as well as an <a href='https://github.com/physics-codes/seminar'>example syllabus</a>.
+  The notebook annotations are aimed to help those who are using the book as instructors and/or students.
+</p>
+
+<p>
+  <h4>
+    A response to my original discuss question revolved around administrative and university department politics on who should and who may teach a course like this.
+    For example, some people mentioned that the computer science department will push back to any course involving programming.
+    How did you over come these hurdles, or any others?
+  </h4>
+</p>
+
+<p>
+  [T] This was not much of an issue for me because Quest is a liberal arts college, and this is essentially the only computer science course they are offering this year.
+</p>
+
+<p>
+  [J] I just sneak it into courses that are already on the books.
+  I have created a course from scratch, and it was a nightmare.
+  Teach it in courses that already have a number.
+  But that's not a longterm solution.
+  But it's better that you pilot it in an existing course or a topics course.
+</p>
+
+<p>
+  <emth> How would other people go about piloting a course like this?
+    For example, I'm just a graduate student, would me getting a faculty to sponsor to run a special topics course be the only way to get started?
+  </emth>
+</p>
+
+<p>
+  [J] If someone at UBC wanted to offer this and weren't me (with teaching something or with rights), teach it as a variable special topics.
+  Or find a course that is related, and offer to teach a lap with them.  You can sell the SWC stuff as a lab to an an existing course.
+  This would also provide the added benefit of getting you paid as a TA, for example.
+</p>
+
+<p>
+  <h4>
+    How do we turn the SWC material into a course that fits into 30 weeks.
+    Should we try to expand the current material into 30 weeks, or offer 2 half semester courses.
+    One of the concerns Bill Mills brought up was a 30 week course could too easily evolve/devolve into a Machine Learnig/Data science course.
```

Comment:
definitely devolve :) - more generally, the point I was making is that people usually overshoot what they can cram into a computing course.